sydent/threepid/bind.py

```diff
@@ -13,22 +13,42 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import collections
+import json
+import logging
+import random
+
+import twisted
+from twisted.internet import defer
+from twisted.names import client, dns
+from twisted.names.error import DNSNameError
+from twisted.web.client import Agent, FileBodyProducer
+from twisted.web.http_headers import Headers
+from sydent.db.callbacks import CallbackStore
 
 from sydent.db.valsession import ThreePidValSessionStore
 from sydent.db.threepid_associations import LocalAssociationStore
-from sydent.validators import InvalidSessionIdException, IncorrectClientSecretException
+from sydent.http.httpsclient import SydentPolicyForHTTPS
 
 from sydent.util import time_msec
 from sydent.threepid.assocsigner import AssociationSigner
 
 from sydent.threepid import ThreepidAssociation
 
+from StringIO import StringIO
+
+
+logger = logging.getLogger(__name__)
+
+
 class ThreepidBinder:
     # the lifetime of a 3pid association
     THREEPID_ASSOCIATION_LIFETIME_MS = 100 * 365 * 24 * 60 * 60 * 1000
 
     def __init__(self, sydent):
         self.sydent = sydent
+        self.store = CallbackStore(self.sydent)
+        self.agent = Agent(twisted.internet.reactor, SydentPolicyForHTTPS(self.sydent))
```

Comment:
I think you probably don't want SydentPolicyForHTTPS here: that loads the root cert from the config which will be the one used for validating IS certs (which might be a special CA cert). I think the normal one in twisted is BrowerLikePolicyForHTTPS or something.